pkg/dbclient/client.go

```diff
@@ -567,24 +567,42 @@ func (pc *client) UserExists(userName string) (bool, error) {
 	return exists, nil
 }
 
-func (pc *client) DeleteUser(username, reassignToUser string) error {
-	db := pc.DB
+func (pc *client) DeleteUser(dbName, username, reassignToUser string) error {
```

Comment:
This is a long series of async calls made into the database, can you use a context.Context and switch the calls to context versions?
```suggestion
func (pc *client) DeleteUser(ctx context.Context, dbName, username, reassignToUser string) error {
```

It makes the system work far more gracefully. We can pass a context into this client now. On shutdown, we notify the client that a shutdown is happening and it needs to stop what it is doing. Imagine if we're in the middle of a migration and we just `os.Exit(1)` on the database. The database is now in an unknown state and we may not be able to fully recovery it.